0.13.7 BenchmarkDotNet.Autogenerated project fails build

In VS2022 with a .NET 4.8.1 project in release cfg, the Autogenerated project is failing to build using 0.13.7 with the following errors, but it works fine with 0.13.6.  The generated project files appear to be the same between the package versions as well so something else is going on.  Interestingly, adding an explicit PackageReference for BenchmarkDotNet to the BenchmarkDotNet.Autogenerated.csproj makes it work.

```
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(136,48): error CS0234: The type or namespace name 'Engines' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(328,48): error CS0234: The type or namespace name 'Engines' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(245,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(437,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(270,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(462,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(280,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(472,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(305,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
8e7c92b4-3761-43e1-b9bd-ee799e381dbf.notcs(497,53): error CS0234: The type or namespace name 'Portability' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [BenchmarkDotNet.Autogenerated.csproj]
```

Note: Pathing was stripped for brevity.

--- a/src/BenchmarkDotNet/Toolchains/ToolchainExtensions.cs
+++ b/src/BenchmarkDotNet/Toolchains/ToolchainExtensions.cs
@@ -28,19 +28,17 @@ namespace BenchmarkDotNet.Toolchains
                 : GetToolchain(
                     job.ResolveValue(EnvironmentMode.RuntimeCharacteristic, EnvironmentResolver.Instance),
                     descriptor,
-                    job.HasValue(InfrastructureMode.NuGetReferencesCharacteristic) || job.HasValue(InfrastructureMode.BuildConfigurationCharacteristic));
+                    job.HasValue(InfrastructureMode.NuGetReferencesCharacteristic)
+                    || job.HasValue(InfrastructureMode.BuildConfigurationCharacteristic)
+                    || job.HasValue(InfrastructureMode.ArgumentsCharacteristic));
 
         internal static IToolchain GetToolchain(this Runtime runtime, Descriptor descriptor = null, bool preferMsBuildToolchains = false)
         {
             switch (runtime)
             {
                 case ClrRuntime clrRuntime:
-                    if (!preferMsBuildToolchains && RuntimeInformation.IsFullFramework &&
-                        // If dotnet SDK is not installed, we use RoslynToolchain.
-                        (!HostEnvironmentInfo.GetCurrent().IsDotNetCliInstalled()
-                        // Integration tests take too much time, because each benchmark run rebuilds the test suite and BenchmarkDotNet itself.
-                        // To reduce the total duration of the CI workflows, we just use RoslynToolchain.
-                        || XUnitHelper.IsIntegrationTest.Value))
+                    if (!preferMsBuildToolchains && RuntimeInformation.IsFullFramework
+                        && RuntimeInformation.GetCurrentRuntime().MsBuildMoniker == runtime.MsBuildMoniker)
                     {
                         return RoslynToolchain.Instance;
                     }

